### ISSUE TITLE
error in "traditional" Gumbel Softmax trick

### ISSUE THREAD

**Issue by balcilar** [nec-research/tf-imle#1]
Thanks for the Discete VAE demo.
Seems there is small error in  "def sample_gumbel(self, shape, eps=1e-20): " function.
U is not assigned. It seems to be U= tf.random.uniform(shape, minval=0, maxval=1) ?

**Comment by mniepert** [comment]
Yes thanks a lot. This was a problem I probably introduced when cleaning up the code. I’ll push the fix tomorrow.

**Comment by mniepert** [comment]
Fixed the typo

**Comment by balcilar** [comment]
Thanks. But this time seems temperature is treated as it is eps parameter in "def sample_gumbel(self, shape, eps=1e-20)" function.
It seems to be called by 
y = logits + self.sample_gumbel(tf.shape(logits)) ?

**Comment by mniepert** [comment]
Fixed as well. The temperature as second argument is not correct. It is not used in sample_gumbel()
